src/front/spv/flow.rs

```diff
@@ -346,7 +346,10 @@ impl FlowGraph {
                     _ => return Err(Error::InvalidTerminator),
                 };
 
-                Ok(vec![crate::Statement::Loop { body, continuing }])
+                let mut result = vec![crate::Statement::Loop { body, continuing }];
+                result.extend(self.naga_traverse(merge_node_index, None)?);
```

Comment:
We need to revise the return type here to avoid double-allocation at some point. Alternatively, we'd first get the vector from the function, and then insert our statement in the 0th place.